### PR TITLE
Resolves #59015, Scheduler: Add support for EBS types t3, r5, & z1d

### LINKED ISSUE #59015: Number of NVMe disks attachable is lower than max predefined count for EBS
Kubernetes's scheduler defined a default limit of 39 volumes for EBS disks per node (`DefaultMaxEBSVolumes`) but this limit is far lower  for NVMe disks (something like 25-28). We need to find out why and if we should document this.

### PR DESCRIPTION
**Pull request by cuppett**
The new t3, r5, r5d and z1 need matched as well according to this:

https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/volume_limits.html

T3, C5, C5d, M5, M5d, R5, R5d, and z1d instances support a maximum of 28 attachments, and every instance has at least one network interface attachment. If you have no additional network interface attachments on these instances, you could attach 27 EBS volumes.

fixes #59015

### CODE REVIEW

File: pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -378,7 +378,7 @@ func getMaxVolumeFunc(filterName string) func(node *v1.Node) int {
 }
 
 func getMaxEBSVolume(nodeInstanceType string) int {
-	if ok, _ := regexp.MatchString("^[cm]5.*", nodeInstanceType); ok {
+	if ok, _ := regexp.MatchString("^[cmr]5.*|t3|z1d", nodeInstanceType); ok {
```
Comment (misterikkit) [inline pkg/scheduler/algorithm/predicates/predicates.go]: Having the same regex in two places that need to be kept in sync makes me uneasy. Also, for someone who is not in the know, how would I learn whether this regex is correct? Are there test cases for the supported AWS volume types?

**Review by misterikkit — COMMENTED** [review]
I'll leave it to sigs aws and storage to approve of the regex change.

I would prefer to see a git commit message as descriptive as your PR description, and I would also like for the title/top line to be more descriptive. e.g. instead of "resolves #xyz" it could be something like, "scheduler: add support for EBS types t3, r5, & z1d"

File: pkg/scheduler/algorithm/predicates/predicates.go (hunk 1 above)
Comment (cuppett) [reply pkg/scheduler/algorithm/predicates/predicates.go]: There is an existing test from the previous PR #66397 for m5/c5. Updated commit description and title. Thanks!

**Pull request closed without merging**